src/libYARP_sig/src/yarp/sig/ImageFile.cpp

```diff
@@ -275,30 +306,110 @@ static bool ImageReadRGB_PxM(ImageOf<PixelRgb> &img, const char *filename)
     return true;
 }
 
-static bool ImageReadFloat_PxM(ImageOf<PixelFloat>& dest, const std::string& filename)
+#if defined (YARP_HAS_ZLIB)
+bool ImageReadFloat_CompressedHeaderless(ImageOf<PixelFloat>& dest, const std::string& filename)
 {
-    FILE *fp = fopen(filename.c_str(), "rb");
+    FILE* fp = fopen(filename.c_str(), "rb");
     if (fp == nullptr) {
         return false;
     }
 
     size_t br = 0;
-    size_t size_ = sizeof(float);
     size_t count_ = 0;
 
+    //get the file size
+    fseek(fp, 0, SEEK_END);
+    size_t sizeDataCompressed = ftell(fp);
+    rewind(fp);
+
+    //read the compressed data
+    char* dataReadInCompressed = new char[sizeDataCompressed];
+    br = fread(dataReadInCompressed, 1, sizeDataCompressed, fp);
+    fclose(fp);
+
+    if (br != sizeDataCompressed) { yError() << "problems reading file!"; return false; }
+
+    size_t h = ((size_t*)(dataReadInCompressed))[0]; //byte 0
+    size_t w = ((size_t*)(dataReadInCompressed))[1]; //byte 8, because size_t is 8 bytes long
+    size_t hds = 2* sizeof(size_t); //16 bytes
+
+    dest.resize(w, h);
+    unsigned char* destbuff = dest.getRawImage();
+    //this is the size of the image
+    size_t sizeDataUncompressed = dest.getRawImageSize();
+    //this is the size of the buffer. Extra space is required for temporary operations (I choose arbitrarily *2)
+    size_t sizeDataUncompressedExtra = sizeDataUncompressed*2;
+
+    char* dataUncompressed = new char[sizeDataUncompressedExtra];
+
+    int z_result = uncompress((Bytef*) dataUncompressed, (uLongf*)&sizeDataUncompressedExtra, (const Bytef*)dataReadInCompressed+ hds, sizeDataCompressed- hds);
+    switch (z_result)
+    {
+    case Z_OK:
+        break;
+
+    case Z_MEM_ERROR:
+        yCError(IMAGEFILE, "zlib compression: out of memory");
+        delete[] dataUncompressed;
+        return false;
+        break;
+
+    case Z_BUF_ERROR:
+        yCError(IMAGEFILE, "zlib compression: output buffer wasn't large enough");
+        delete[] dataUncompressed;
+        return false;
+        break;
+
+    case Z_DATA_ERROR:
+        yCError(IMAGEFILE, "zlib compression: file contains corrupted data");
+        delete[] dataUncompressed;
+        return false;
+        break;
+    }
+
+    //here I am copy only the size of the image, obviously the extra space is not needed anymore.
+    for (size_t i=0; i< sizeDataUncompressed; i++)
+    {
+        destbuff[i] = dataUncompressed[i];
+    }
+
+    delete [] dataUncompressed;
+    return true;
+}
+#endif
+
+bool ImageReadFloat_PlainHeaderless(ImageOf<PixelFloat>& dest, const std::string& filename)
+{
+    FILE *fp = fopen(filename.c_str(), "rb");
+    if (fp == nullptr) {
+        return false;
+    }
+
     size_t dims[2];
-    if (fread(dims, sizeof(dims), 1, fp) > 0)
+    if (fread(dims, sizeof(dims), 1, fp) <= 0)
```

Comment:
Codacy found an issue: [Checking if unsigned expression 'fread(dims,sizeof(dims),1,fp)' is less than zero.](https://app.codacy.com/manual/robotology/yarp/pullRequest?prid=6165005)